src/test/java/org/sagebionetworks/bridge/models/files/FileReferenceTest.java

```diff
@@ -0,0 +1,93 @@
+package org.sagebionetworks.bridge.models.files;
+
+import static org.sagebionetworks.bridge.TestConstants.GUID;
+import static org.sagebionetworks.bridge.TestConstants.TIMESTAMP;
+import static org.sagebionetworks.bridge.config.Environment.PROD;
+import static org.sagebionetworks.bridge.config.Environment.UAT;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNull;
+
+import com.fasterxml.jackson.databind.JsonNode;
+
+import org.mockito.Mockito;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.testng.PowerMockTestCase;
+import org.testng.annotations.Test;
+
+import org.sagebionetworks.bridge.config.BridgeConfig;
+import org.sagebionetworks.bridge.config.BridgeConfigFactory;
+import org.sagebionetworks.bridge.json.BridgeObjectMapper;
+
+import nl.jqno.equalsverifier.EqualsVerifier;
+
+@PrepareForTest({BridgeConfigFactory.class})
+public class FileReferenceTest extends PowerMockTestCase {
+    
+    private BridgeConfig config;
+    
+    @Test
+    public void equalsVerifier() {
+        EqualsVerifier.forClass(FileReference.class).allFieldsShouldBeUsed().verify();
+    }
+    
+    @Test
+    public void nullGuid() throws Exception {
+        FileReference ref = new FileReference(null, TIMESTAMP);
+        assertNull(ref.getGuid());
+        assertEquals(ref.getCreatedOn(), TIMESTAMP);
+        assertNull(ref.getHref());
+    }
+    
+    @Test
+    public void nullCreatedOn() throws Exception {
+        FileReference ref = new FileReference(GUID, null);
+        assertEquals(ref.getGuid(), GUID);
+        assertNull(ref.getCreatedOn());
+        assertNull(ref.getHref());
+    }
+    
+    @Test
+    public void canSerialization() throws Exception {
+        String href = "http://" + BridgeConfigFactory.getConfig()
+                .getHostnameWithPostfix("docs") + "/" + GUID + "." + TIMESTAMP.getMillis();
+        
+        FileReference ref = new FileReference(GUID, TIMESTAMP);
+        
+        JsonNode node = BridgeObjectMapper.get().valueToTree(ref);
+        assertEquals(node.get("guid").textValue(), GUID);
+        assertEquals(node.get("createdOn").textValue(), TIMESTAMP.toString());
+        assertEquals(node.get("type").textValue(), "FileReference");
```

Comment:
We should also check that href is being serialized correctly.

Reply:
Added